lib/ucfg.rb

```diff
@@ -43,20 +43,27 @@ def self.validate_recursively(config, schema, config_path)
     # fails if property is provided as other type
     config.each do |key, value|
       if schema["properties"].key?(key) && schema["properties"][key].key?("type")
-        if schema["properties"][key]["type"] != value_type(value)
+        if schema["properties"][key]["type"].include?(value_type(value))
+         valid = true
+        elsif value.nil?
+          valid = false
+          errors << "Property `#{(config_path + [key]).join('.')}` must be of type `#{(schema['properties'][key]['type']).first}` or `#{(schema['properties'][key]['type']).last}` (provided `#{value_type(value)}`)"
```

Comment:
In the case of value being null, we can't expect there to be 2 types. I have modified one of the specs and you would need to make changes accordingly on this line.

Reply:
Done